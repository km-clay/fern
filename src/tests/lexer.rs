use super::super::*;
#[test]
fn lex_simple() {
	let input = "echo hello world";
	let tokens: Vec<_> = LexStream::new(input, LexFlags::empty()).collect();

	insta::assert_debug_snapshot!(tokens)
}
#[test]
fn lex_redir() {
	let input = "echo foo > bar.txt";
	let tokens: Vec<_> = LexStream::new(input, LexFlags::empty()).collect();

	insta::assert_debug_snapshot!(tokens)
}
#[test]
fn lex_redir_fds() {
	let input = "echo foo 1>&2";
	let tokens: Vec<_> = LexStream::new(input, LexFlags::empty()).collect();

	insta::assert_debug_snapshot!(tokens)
}
#[test]
fn lex_quote_str() {
	let input = "echo \"foo bar\" biz baz";
	let tokens: Vec<_> = LexStream::new(input, LexFlags::empty()).collect();

	insta::assert_debug_snapshot!(tokens)
}
#[test]
fn lex_with_keywords() {
	let input = "if true; then echo foo; fi";
	let tokens: Vec<_> = LexStream::new(input, LexFlags::empty()).collect();

	insta::assert_debug_snapshot!(tokens)
}

#[test]
fn lex_multiline() {
	let input = "echo hello world\necho foo bar\necho boo biz";
	let tokens: Vec<_> = LexStream::new(input, LexFlags::empty()).collect();

	insta::assert_debug_snapshot!(tokens)
}
