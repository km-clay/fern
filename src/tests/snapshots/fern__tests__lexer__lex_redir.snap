---
source: src/tests/lexer.rs
expression: tokens
---
[
    Ok(
        Tk {
            class: SOI,
            span: Span {
                range: 0..0,
                source: "echo foo > bar.txt",
            },
            flags: TkFlags(
                0x0,
            ),
        },
    ),
    Ok(
        Tk {
            class: Str,
            span: Span {
                range: 0..4,
                source: "echo foo > bar.txt",
            },
            flags: TkFlags(
                IS_CMD | BUILTIN,
            ),
        },
    ),
    Ok(
        Tk {
            class: Str,
            span: Span {
                range: 5..8,
                source: "echo foo > bar.txt",
            },
            flags: TkFlags(
                0x0,
            ),
        },
    ),
    Ok(
        Tk {
            class: Redir,
            span: Span {
                range: 9..10,
                source: "echo foo > bar.txt",
            },
            flags: TkFlags(
                0x0,
            ),
        },
    ),
    Ok(
        Tk {
            class: Str,
            span: Span {
                range: 11..18,
                source: "echo foo > bar.txt",
            },
            flags: TkFlags(
                0x0,
            ),
        },
    ),
    Ok(
        Tk {
            class: EOI,
            span: Span {
                range: 18..18,
                source: "echo foo > bar.txt",
            },
            flags: TkFlags(
                0x0,
            ),
        },
    ),
]
